juniper_codegen/src/util/tracing.rs

```diff
@@ -0,0 +1,352 @@
+use std::{collections::HashMap, mem, str::FromStr};
+
+use proc_macro2::TokenStream;
+use proc_macro_error::abort;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    spanned::Spanned as _,
+    token,
+};
+
+#[derive(Debug, Default)]
+pub struct Attr {
+    /// Optional span rename, if `None` method name should be used instead.
+    name: Option<syn::LitStr>,
+
+    /// Overwritten `level` of span generated, if `None` `Level::INFO` should be used.
+    level: Option<syn::LitStr>,
```

Comment:
@ArsileLuci also, why parse all the attribute arguments by ourselves? Cannot we piggyback everything to upstream (`tracing` crate)?

Reply:
@tyranron [`span!(...)`](https://docs.rs/tracing/0.1.26/tracing/macro.span.html) macro is pretty strict on argument order, and we need to cleanup arguments specific for `juniper`, so a little of extra work here will allow to forget about argument order.